code/modules/plumbing/plumbers/pill_press.dm

```diff
@@ -43,26 +49,32 @@
 /obj/machinery/plumbing/pill_press/process()
 	if(machine_stat & NOPOWER)
 		return
-	if(reagents.total_volume >= pill_size)
-		var/obj/item/reagent_containers/pill/P = new(src)
-		reagents.trans_to(P, pill_size)
-		P.name = pill_name
-		stored_pills += P
-		if(pill_number == RANDOM_PILL_STYLE)
-			P.icon_state = "pill[rand(1,21)]"
-		else
-			P.icon_state = "pill[pill_number]"
-		if(P.icon_state == "pill4") //mirrored from chem masters
-			P.desc = "A tablet or capsule, but not just any, a red one, one taken by the ones not scared of knowledge, freedom, uncertainty and the brutal truths of reality."
-	if(stored_pills.len)
+	if(reagents.total_volume >= current_volume)
+		if (product == "pill")
+			var/obj/item/reagent_containers/pill/P = new(src)
+			reagents.trans_to(P, current_volume)
+			P.name = trim("[product_name] pill")
```

Comment:
Combined with your changes on the UI, it means that you can no longer call pills Pilliam. Instead it’ll become Pilliam pill. Same way if someone uses some other form of the word pill

Reply:
Yeah, I added "pill" or "patch" suffix to the end of the product name and text with that suffix to the UI. It just makes it easier to handle in the code, also makes naming consistent with Chem Master.